File: app/src/main/java/djordjeh/architecture/mvp/ui/tasks/HomeActivityModule.java
package djordjeh.architecture.mvp.ui.tasks;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import djordjeh.architecture.mvp.dagger.scope.FragmentScope;

@Module
public abstract class HomeActivityModule {

    @FragmentScope
    @ContributesAndroidInjector(modules = TasksModule.class)
    abstract TasksFragment contributeTasksFragment();
}

File: app/src/main/java/djordjeh/architecture/mvp/ui/tasks/HomeActivityModule.kt
package djordjeh.architecture.mvp.ui.tasks

import dagger.Module
import dagger.android.ContributesAndroidInjector
import djordjeh.architecture.mvp.dagger.scope.FragmentScope

@Module
abstract class HomeActivityModule {

    @FragmentScope
    @ContributesAndroidInjector(modules = [TasksModule::class])
    internal abstract fun contributeTasksFragment(): TasksFragment
}
